tests/export_batch_if_test.cc.cu

```diff
@@ -0,0 +1,188 @@
+                            buffer0.scores_ptr(), buffer0_ev.keys_ptr(),
+                            buffer0_ev.values_ptr(), buffer0_ev.scores_ptr(),
+                            d_cnt, stream, true, false);
+    table_size = table->size(stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    MERLIN_EXPECT_TRUE(table_size == n0, "Invalid table size.");
+  }
+
+  test_util::KVMSBuffer<i64, f32, u64> buffer1;
+  buffer1.Reserve(n1, dim, stream);
+  buffer1.ToRange(n0, 1, stream);
+  buffer1.Setscore((u64)30, stream);
+  {
+    test_util::KVMSBuffer<i64, f32, u64> buffer1_ev;
+    buffer1_ev.Reserve(n0, dim, stream);
+    buffer1_ev.ToZeros(stream);
+    // table->insert_or_assign(n1, buffer1.keys_ptr(), buffer1.values_ptr(),
+    //                         buffer1.scores_ptr(), stream, true, false);
+    table->insert_and_evict(n0, buffer1.keys_ptr(), buffer1.values_ptr(),
+                            buffer1.scores_ptr(), buffer1_ev.keys_ptr(),
+                            buffer1_ev.values_ptr(), buffer1_ev.scores_ptr(),
+                            d_cnt, stream, true, false);
+    table_size = table->size(stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+  }
+
+  test_util::KVMSBuffer<i64, f32, u64> buffer2;
+  buffer2.Reserve(n2, dim, stream);
+  buffer2.ToRange(n0 + n1, 1, stream);
+  buffer2.Setscore((u64)45, stream);
+  {
+    test_util::KVMSBuffer<i64, f32, u64> buffer2_ev;
+    buffer2_ev.Reserve(n0, dim, stream);
+    buffer2_ev.ToZeros(stream);
+    // table->insert_or_assign(n2, buffer2.keys_ptr(), buffer2.values_ptr(),
+    //                         buffer2.scores_ptr(), stream, true, false);
+    table->insert_and_evict(n0, buffer2.keys_ptr(), buffer2.values_ptr(),
+                            buffer2.scores_ptr(), buffer2_ev.keys_ptr(),
+                            buffer2_ev.values_ptr(), buffer2_ev.scores_ptr(),
+                            d_cnt, stream, true, false);
+    table_size = table->size(stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    printf("final size: %zu, capacity: %zu\n", table_size, table->capacity());
+  }
+
+  size_t h_cnt = 0;
+  size_t h_cnt2 = 0;
+
+  table->size_if<ExportIfPredFunctor>(pattern, threshold, d_cnt, stream);
+  CUDA_CHECK(cudaMemcpyAsync(&h_cnt, d_cnt, sizeof(size_t),
+                             cudaMemcpyDeviceToHost, stream));
+  CUDA_CHECK(cudaStreamSynchronize(stream));
+  printf("---> check h_cnt from size_if kernel: %zu\n", h_cnt);
+
+  test_util::KVMSBuffer<i64, f32, u64> buffer_out;
+  buffer_out.Reserve(h_cnt, dim, stream);
+  buffer_out.ToZeros(stream);
+
+  CUDA_CHECK(cudaMemsetAsync(d_cnt, 0, sizeof(size_t), stream));
+  CUDA_CHECK(cudaStreamSynchronize(stream));
+
+  bool use_pin = true;
+
+  uint64_t t0 = test_util::getTimestamp();
+  cudaEvent_t start, stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+  cudaEventRecord(start);
+  table->export_batch_if<ExportIfPredFunctor>(
```

Comment:
There are total three kernel templates, and has each kernel been tested? If not is it necessary to test each kernel.

Reply:
I've tested them seperately, but not add it into the tests case, since it's an internal option not for public API.